carbon-design-system/carbon-addons-iot-react
# fix(table): create table action on search expand

Closes #3341 

**Summary**

- Create new table action on search expand, deprecated previous one.

**Change List (commits, features, bugs, etc)**

- Add table action that triggers on search open/close
- Add deprecation notice to previous search expand callback - `onExpand`
- Add deprecation notice to unused `onChange` callback in toolbar search
- Update test
- Update docs

**Acceptance Test (how to verify the PR)**

- Go to [this story](https://deploy-preview-3588--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-searching)
- Open `Actions` tab
- Focus on search input
- Verify that `onSearchExpand` action has fired with event class and `true` parameters
- Clear search input and move focus away from search bar
- Verify that `onSearchExpand` action has fired with event class and `false` parameters
- In the Knobs tab change type of table to `Table`
- Verify that `onSearchExpand` triggers on the same action as in `StatefulTable`

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
